cmd/ocm-backplane/login/login.go

```diff
@@ -284,13 +284,17 @@ func GetRestConfig(bp config.BackplaneConfiguration, clusterID string) (*rest.Co
 
 // GetRestConfigAsUser returns a client-go *rest.Config like GetRestConfig, but supports configuring an
 // impersonation username. Commonly, this is "backplane-cluster-admin"
-func GetRestConfigAsUser(bp config.BackplaneConfiguration, clusterID, username string) (*rest.Config, error) {
+// best practice would be to add at least one elevationReason in order to justity the impersonation
+func GetRestConfigAsUser(bp config.BackplaneConfiguration, clusterID, username string, elevationReason ...string) (*rest.Config, error) {
 	cfg, err := GetRestConfig(bp, clusterID)
 	if err != nil {
 		return nil, err
 	}
 
-	cfg.Impersonate = rest.ImpersonationConfig{UserName: username}
+	cfg.Impersonate = rest.ImpersonationConfig{
+		UserName: username,
+		Extra: map[string][]string{"reason": elevationReason},
```

Comment:
What if adding the `reason` only for `admin` users ? 
or check the length of `elevationReason `  before adding the reason

Reply:
> or check the length of elevationReason  before adding the reason
+1, we should only add the `reason` key if there are elevationReasons supplied

Reply:
done